lusidtools/cocoon/cocoon_printer.py

```diff
@@ -0,0 +1,171 @@
+import pandas as pd
+import numpy as np
+
+
+def format_instruments_response(response: dict, file_type: str = "instruments") -> (
+pd.DataFrame, pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from instrument requests and returns successful, failed and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :param str file_type: file_type of Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    :return: pd.DataFrame: Failed responses that LUSID rejected
+    """
+    items_success, items_failed, items_error = [], [], []
+    if "success" not in response[file_type].keys() or \
+            "errors" not in response[file_type].keys():
+        raise ValueError("Didn't recognise response shape. Must include 'errors', and 'success' keys")
+
+    # get successes and failures
+    for batch in response[file_type]["success"]:
+        [items_success.append(batch.values[key].name) for key in batch.values.keys()]
+        if len(response[file_type]["success"][0].failed) > 0:
+            [items_failed.append(batch.failed[key].name) for key in batch.failed.keys()]
+        else:
+            items_failed = []
+
+    # get API exception
+    if len(response[file_type]["errors"]) > 0:
+        status = [item.status for item in response[file_type]["errors"]]
+        reason = [item.reason for item in response[file_type]["errors"]]
+        items_error = np.array([reason, status]).T.tolist()
+        errors = pd.DataFrame(items_error, columns=["error_items", "status"])
+    else:
+        errors = pd.DataFrame([], columns=["error_items", "status"])
+
+    return (pd.DataFrame(items_success, columns=["successful items"]),
+            errors,
+            pd.DataFrame(items_failed, columns=["failed_items"]))
+
+
+def format_portfolios_response(response: dict, file_type: str = "portfolios") -> (pd.DataFrame, pd.DataFrame):
+    """
+    This function unpacks a response from portfolio requests and returns successful and errored statuses for
+    request constituents.
+    :param dict response: response from Lusid-python-tools
+    :param str file_type: file_type of Lusid-python-tools
+    :return: pd.DataFrame: Successful calls from request
+    :return: pd.DataFrame: Error responses from request that fail (APIExceptions: 400 errors)
+    """
+    items_success, items_error = [], []
+
+    if "success" not in response[file_type].keys() or \
+            "errors" not in response[file_type].keys():
+        raise ValueError("Didn't recognise response shape. Must include 'errors', and 'success' keys")
```

Comment:
Can we delegate this to another function? As we call it for each of these 5 functions, so if we need to change it you have to do it in 5 places atm!

Reply:
`check_errors_and_success()`